lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@riccardonar I don't see how that's true. The server is rejecting the option -- not the client

IE, if you're connecting from a centos 6 install out of the box... it's going to have the 8.4 client. PGSQL_LIBPQ_VERSION = 8.4.

I can't use that value to determine what options to use, as I need to know what options the SERVER supports.

IE, i'd be checking logic againstn 8.4 when the server is running 9.3

The only way using PGSQL_LIBPQ_VERSION  would work is if you ensure you are manually upgrading the CLIENT on your web severs to match the same release as your SERVER. That seems like a bad idea.

Please clarify?